test/client/fixtures/sandbox/code-instrumentation/location-test.js

```diff
@@ -157,6 +192,83 @@ test('special pages (GH-339)', function () {
     destLocation.forceLocation(storedForcedLocation);
 });
 
+test('should add the trailing slash to location.href using href.set(), assign() and replace() functions if url consist of origin (GH-1426)', function () {
+    var setHref = function (url) {
+        this.href = url;
+    };
+
+    function proxyUrl (url) {
+        urlUtils.getProxyUrl(url);
+    }
+
+    function proxyUrlTrailingSlash (urlCase) {
+        var proxiedUrl = urlUtils.getProxyUrl(urlCase.url);
+
+        return proxiedUrl + (urlCase.shoudAddTrailingSlash ? '/' : '');
+    }
+
+    var windowMock = {
+        location: {
+            href:     '',
+            replace:  setHref,
+            assign:   setHref,
+            toString: function () {
+                return proxyUrl(this.location.href);
+            }
+        }
+    };
+
+    windowMock.top = windowMock;
+
+    var locationWrapper = new LocationWrapper(windowMock);
+
+    function checkTrailingSlash (urlCases) {
+        urlCases.forEach(function (urlCase) {
+            locationWrapper.href = urlCase.url;
+            strictEqual(windowMock.location.href, proxyUrlTrailingSlash(urlCase));
+
+            locationWrapper.replace(urlCase.url);
+            strictEqual(windowMock.location.href, proxyUrlTrailingSlash(urlCase));
+
+            locationWrapper.assign(urlCase.url);
+            strictEqual(windowMock.location.href, proxyUrlTrailingSlash(urlCase));
+        });
+    }
+
+    checkTrailingSlash(URL_TEST_CASES);
+});
+
+test('should add the trailing slash to location when navigateTo() is called (GH-1426)', function () {
```

Comment:
should ensure a trailing slash on page navigation using `hammerhead.navigateTo` method (GH-1426)